hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/StandardizedAttributeValidator.java

```diff
@@ -0,0 +1,139 @@
+    private final GlobalDynamicProperties dynamicProperties;
+
+    @Inject
+    public StandardizedAttributeValidator(
+            @NonNull final LongSupplier consensusSecondNow,
+            @NonNull final @CompositeProps PropertySource properties,
+            @NonNull final GlobalDynamicProperties dynamicProperties) {
+        this.maxEntityLifetime = properties.getLongProperty(ENTITIES_MAX_LIFETIME);
+        this.consensusSecondNow = consensusSecondNow;
+        this.dynamicProperties = requireNonNull(dynamicProperties);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void validateKey(@NonNull final Key key) {
+        validateKeyAtLevel(key, 1);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void validateExpiry(long expiry) {
+        final var now = consensusSecondNow.getAsLong();
+        final var expiryGivenMaxLifetime = now + maxEntityLifetime;
+        validateTrue(expiry > now && expiry <= expiryGivenMaxLifetime, INVALID_EXPIRATION_TIME);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void validateAutoRenewPeriod(long autoRenewPeriod) {
+        validateTrue(
+                autoRenewPeriod >= dynamicProperties.minAutoRenewDuration()
+                        && autoRenewPeriod <= dynamicProperties.maxAutoRenewDuration(),
+                AUTORENEW_DURATION_NOT_IN_RANGE);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void validateMemo(@NonNull final String memo) {
+        final var raw = memo.getBytes(StandardCharsets.UTF_8);
+        if (raw.length > dynamicProperties.maxMemoUtf8Bytes()) {
+            throw new HandleException(MEMO_TOO_LONG);
+        } else if (contains(raw, (byte) 0)) {
+            throw new HandleException(INVALID_ZERO_BYTE_IN_STRING);
+        }
+    }
+
+    private void validateKeyAtLevel(@NonNull final Key key, final int level) {
+        if (level > MAX_NESTED_KEY_LEVELS) {
+            throw new HandleException(BAD_ENCODING);
+        }
+        if (!key.hasThresholdKey() && !key.hasKeyList()) {
+            validateSimple(key);
+        } else if (key.hasThresholdKey()
+                && key.thresholdKeyOrThrow().hasKeys()
+                && key.thresholdKeyOrThrow().keysOrThrow().hasKeys()) {
+            key.thresholdKeyOrThrow().keysOrThrow().keysOrThrow().forEach(k -> validateKeyAtLevel(k, level + 1));
+        } else if (key.keyListOrThrow().hasKeys()) {
+            key.keyListOrThrow().keysOrThrow().forEach(k -> validateKeyAtLevel(k, level + 1));
+        }
+    }
+
+    /**
+     * Current behavior is to only invalidate a simple key structure if it has no explicit type. Other validations,
+     * like on the number of bytes in the public key; or on the size of the threshold key; are done elsewhere.
+     *
+     * @param key the key to validate
+     */
+    private void validateSimple(@NonNull final Key key) {
+        if (key.key().kind() == Key.KeyOneOfType.UNSET) {
```

Comment:
We should probably check the `isValid` method as added [here](https://github.com/hashgraph/hedera-services/blob/move-to-account-pbj-type/hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java#L64)?